### PR TITLE
Set pxd minors to 1, pxd disk can't be partitioned

### PR DESCRIPTION
**Pull request by prabirpaul**
**What this PR does / why we need it**:
It's incorrect to set minors to 256 since alloc_disk() was called with minors==1.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pxd.c [hunk 1]
```diff
@@ -1261,7 +1261,7 @@ static int pxd_init_disk(struct pxd_device *pxd_dev, struct pxd_add_ext_out *add
 	snprintf(disk->disk_name, sizeof(disk->disk_name),
 		 PXD_DEV"%llu", pxd_dev->dev_id);
 	disk->major = pxd_dev->major;
-	disk->minors = 256;
+	disk->minors = 1;
```
Comment (sulakshm) [inline pxd.c:1264]: No, this should not be set by us. It is initialized part of alloc_disk argument.

**Comment by sulakshm** [comment]
Would it be possible to put pxd_suspend_io() also same as in 2.10.0? This is in pxd_bio_blkmq.c

File: pxd.c (hunk 1 above)
Comment (prabirpaul) [reply pxd.c:1264]: Kernel 5.15 onwards this is required since alloc_disk doesn't do it. Will move it up.

**Comment by prabirpaul** [comment]
cherry-picked origin/ln/nominors

**Review by sulakshm — APPROVED** [review]

**Pull request merged**